src/trajectory.jl

```diff
@@ -686,13 +687,13 @@ function mh_accept_ratio(
     Horiginal::AbstractVector{<:T},
     Hproposal::AbstractVector{<:T},
 ) where {T<:AbstractFloat}
-    α = min.(1.0, exp.(Horiginal .- Hproposal))
-    accept = _rand(rng) .< α
+    α = min.(one(T), exp.(Horiginal .- Hproposal))
+    accept = _rand(rng, length(Horiginal)) .< α
     return accept, α
 end
 
-_rand(rng::AbstractRNG) = rand(rng)
-_rand(rng::AbstractVector{<:AbstractRNG}) = rand.(rng)
+_rand(rng::AbstractRNG, n_chains::Int) = rand(rng, n_chains)
```

Comment:
This line draws `n_chains` uniform variables from a shared RNG, although the other parts of the sampling process use independent RNGs correctly. This shared RNG for acceptance steps leads to a subtle coupling of vectorised chains, which might cause bias.

Reply:
Maybe force the user to pass a vector of `RNG`s here, and throw an error if not.

Reply:
> This shared RNG for acceptance steps leads to a subtle coupling of vectorised chains, which might cause bias.

I actually discussed with Charles on this and in the end he convinced me that it probably doesn't matter. His argument in brief is below
- What we worry here is the coupling of accepting or rejecting caused by using a shared random seed. That is, e.g., for two chains, `u_1` and `u_2` are coupled in `[u_1, u_2] = rand(2)`.
- This should not be a concern for pseudo random number generators as numbers drawn should be i.i.d (by assuming perfect randomness) anyway.
- If this is a concern, we should also worry that, say, some dimensions of the momentum variables drawn are coupled with some other; but we don't.

Maybe we should get some evidence that this really happens before forcing users to do so?

Reply:
Can you add a comment in the code stating this subtle coupling in case we want to revisit this in the future?  Once that is done I'm ready to merge this PR as is.

Reply:
Just added.